pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -335,10 +351,16 @@ def newpkgcb(pkgId, name, arch):
                     updateinfo_url = urljoin(remote_url, record.location_href)
                     downloader = self.remote.get_downloader(url=updateinfo_url)
                     downloaders.append([downloader.run()])
+
+                elif record.type in COMPS_REPODATA:
+                    comps_url = urljoin(self.remote.url, record.location_href)
```

Comment:
due to kickstarts, we can have multiple repos, `self.remote.url` always points to main repo. `remote_url` can deal with main/sub repos.

Reply:
cool, thanks for explaining this, I will make the change.